driver/src/main/java/org/neo4j/driver/internal/DriverFactory.java

```
@@ -161,15 +161,11 @@ protected static MetricsProvider getOrCreateMetricsProvider(Config config, Clock
         if (metricsAdapter == null) {
             metricsAdapter = config.isMetricsEnabled() ? MetricsAdapter.DEFAULT : MetricsAdapter.DEV_NULL;
         }
-        switch (metricsAdapter) {
-            case DEV_NULL:
-                return DevNullMetricsProvider.INSTANCE;
-            case DEFAULT:
-                return new InternalMetricsProvider(clock, config.logging());
-            case MICROMETER:
-                return MicrometerMetricsProvider.forGlobalRegistry();
-        }
-        throw new IllegalStateException("Unknown or unsupported MetricsAdapter: " + metricsAdapter);
+        return switch (metricsAdapter) {
+            case DEV_NULL -> DevNullMetricsProvider.INSTANCE;
+            case DEFAULT -> new InternalMetricsProvider(clock, config.logging());
+            case MICROMETER -> MicrometerMetricsProvider.forGlobalRegistry();
+        };
```

What happens if `metricAdapter` doesn't match any of the cases?

Should it have a default `throw new IllegalStateException("Unknown or unsupported MetricsAdapter: " + metricsAdapter);`?